addons/html_builder/static/src/builder/plugins/builder_overlay/builder_overlay_plugin.js

```diff
@@ -1,34 +1,247 @@
+        });
+    }
+
+    destroy() {
+        this.removeBuilderOverlay();
+
+        // Remove the listeners.
+        window.removeEventListener("resize", this.update);
+        this.body.removeEventListener("keydown", this.onKeydown);
+        this.body.removeEventListener("mousemove", this.onMouseMoveOrDown);
+        this.body.removeEventListener("mousedown", this.onMouseMoveOrDown);
+        this.scrollingTarget.removeEventListener("scroll", this._onScrollingElementScroll);
+    }
+}
+
+class BuilderOverlay {
+    constructor(overlayElement, overlayTarget, displayHandles) {
+        this.overlayElement = overlayElement;
+        this.overlayTarget = overlayTarget;
+        this.displayHandles = displayHandles;
+        this.handlesWrapperEl = this.overlayElement.querySelector(".o_handles");
+        this.handleEls = this.overlayElement.querySelectorAll(".o_handle");
+    }
+
+    isActive() {
+        return this.overlayElement.classList.contains("oe_active");
+    }
+
+    refreshPosition() {
+        if (!this.isActive()) {
             return;
         }
-        this.removeCurrentOverlay = this.services.overlay.add(BuilderOverlay, {
-            target: toolbox.element,
-            container: this.document.documentElement,
+
+        // TODO transform
+        const targetRect = this.overlayTarget.getBoundingClientRect();
+        Object.assign(this.overlayElement.style, {
+            width: `${targetRect.width}px`,
+            height: `${targetRect.height}px`,
+            top: `${targetRect.y + window.screenY}px`,
+            left: `${targetRect.x + window.screenX}px`,
         });
+        this.handlesWrapperEl.style.height = `${targetRect.height}px`;
+    }
+
+    refreshHandles() {
+        if (!this.displayHandles || !this.isActive()) {
+            return;
+        }
+
+        // TODO improve => by sizing category
+
+        const isMobileView = builderUtils.isMobileView(this.overlayTarget);
+        const isGridOn = this.overlayTarget.classList.contains("o_grid_item");
+        const isGrid = !isMobileView && isGridOn;
+        if (
+            this.overlayTarget.parentNode &&
+            this.overlayTarget.parentNode.classList.contains("row")
+        ) {
+            // Hiding/showing the correct resize handles if we are in grid mode
+            // or not.
+            for (const handleEl of this.handleEls) {
+                const isGridHandle = handleEl.classList.contains("o_grid_handle");
+                handleEl.classList.toggle("d-none", isGrid ^ isGridHandle);
+                // Disabling the vertical resize if we are in mobile view.
+                const isVerticalSizing = handleEl.matches(".n, .s");
+                handleEl.classList.toggle("readonly", isMobileView && isVerticalSizing && isGridOn);
+            }
+        }
+    }
+
+    toggleOverlay(show) {
+        this.overlayElement.classList.add("oe_active", show);
+        this.refreshPosition();
+        this.refreshHandles();
```

Comment:
=== _update ?

Reply:
I would say that, for now, let's keep them separated: when the sizing,... will be implemented, we could still add an update function later to do both at once if we repeat them a lot :slightly_smiling_face: 

And if you meant "using the `_update` function of the plugin after calling `toggleOverlay`", then I prefer doing the refresh here, so the overlay logic is grouped at one place (easier to follow the flow), like each overlay updates itself :+1: